src/psyclone/psyGen.py

```diff
@@ -5335,6 +5393,58 @@ def else_body(self):
             return self._children[2]
         return None
 
+    @staticmethod
+    def create(if_condition, if_body, else_body=None):
+        '''Create an IfBlock instance given valid instances of an
+        if_condition, an if_body and an optional else_body.
+
+        :param if_condition: the PSyIR node containing the if \
+            condition of the if block.
+        :type if_condition: :py:class:`psyclone.psyGen.Node`
+        :param if_body: the PSyIR node containing the if body of \
+            the if block.
+        :type if_body: :py:class:`psyclone.psyGen.Schedule`
+        :param else_body: PSyIR node containing the else body of the \
+            if block of None if there is no else body (defaults to None).
+        :type else_body: :py:class:`psyclone.psyGen.Schedule` or \
+            NoneType
+
+        :returns: an IfBlock instance.
+        :rtype: :py:class:`psyclone.psyGen.IfBlock`
+
+        '''
+        if not isinstance(if_condition, Node):
+            raise GenerationError(
+                "if_condition argument to class IfBlock method create "
+                "should be a PSyIR Node but found '{0}'."
+                "".format(type(if_condition).__name__))
+        if not (isinstance(if_body, list) and all(isinstance(child, Node)
+                                                  for child in if_body)):
+            raise GenerationError(
+                "if_body argument to class IfBlock method create should be "
+                "a list of PSyIR Nodes but it is either not a list or "
+                "one of the list's children is not a Node.")
+        if else_body and not (isinstance(if_body, list) and
+                all(isinstance(child, Node) for child in else_body)):
```

Comment:
pycodestyle: continuation line under-indented for visual indent

Reply:
Oops.